From 54291786967bb1df76b21c6bc30319f19a4f8cdd Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Tue, 26 Nov 2024 16:55:34 -0600
Subject: [PATCH] trivial: msr: fix a crash on recognized AMD CPUs

Any CPUs that are not vulnerable to Sinkclose don't have any known
mitigations needed right now.  Don't crash the MSR plugin for this
scenario.

Fixes: https://github.com/fwupd/fwupd/issues/8161
---
 plugins/msr/fu-msr-plugin.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/plugins/msr/fu-msr-plugin.c b/plugins/msr/fu-msr-plugin.c
index 55c7a06a4..5e449c506 100644
--- a/plugins/msr/fu-msr-plugin.c
+++ b/plugins/msr/fu-msr-plugin.c
@@ -673,6 +673,7 @@ fu_msr_plugin_add_security_attr_amd_hwcr(FuPlugin *plugin, FuSecurityAttrs *attr
 	FuDevice *device = fu_plugin_cache_lookup(plugin, "cpu");
 	gboolean sinkclose_vuln = FALSE;
 	g_autoptr(FwupdSecurityAttr) attr1 = NULL;
+	g_auto(GStrv) mitigations = NULL;
 
 	/* this MSR is only valid for a subset of AMD CPUs */
 	if (fu_cpu_get_vendor() != FU_CPU_VENDOR_AMD)
@@ -685,8 +686,10 @@ fu_msr_plugin_add_security_attr_amd_hwcr(FuPlugin *plugin, FuSecurityAttrs *attr
 	if (device != NULL) {
 		const gchar *needed =
 		    fu_device_get_metadata(device, FU_DEVICE_METADATA_CPU_MITIGATIONS_REQUIRED);
-		g_auto(GStrv) mitigations = NULL;
-		mitigations = g_strsplit(needed, ",", -1);
+		if (needed != NULL)
+			mitigations = g_strsplit(needed, ",", -1);
+	}
+	if (mitigations != NULL) {
 		for (guint i = 0; mitigations[i] != NULL; i++) {
 			/* check for sinkclose vulnerability */
 			if (g_strcmp0(mitigations[i], "sinkclose") == 0) {
-- 
2.43.0

